checker/src/org/checkerframework/checker/regex/classic/RegexTransfer.java

```diff
@@ -89,6 +90,8 @@ public RegexTransfer(RegexAnalysis analysis) {
                         thenStore.insertValue(firstParam, regexAnnotation);
                     }
                 } catch (FlowExpressionParseException e) {
+                    // report errors here
+                    ((RegexChecker) (analysis.getTypeFactory().getContext())).report(e.getResult(), n);
                     assert false;
```

Comment:
To me the assert false says this should never happen, and I am just adding a second way for this error to be reported. I don't know enough about the design of the Regex Checker to determine that it's safe to remove it. At a glance, this method is parsing an expression (`"#1"`) that should always be valid for the given method (`isRegex`), so it does look like this should never fail unless there is a bug in the implementation, and that we should keep the assert.